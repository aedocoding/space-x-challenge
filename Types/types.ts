export interface ILaunchListItem {
  name: string;
  details: string;
  image: string;
}
export type ScreenType = "LaunchList" | "CreateLaunch";
