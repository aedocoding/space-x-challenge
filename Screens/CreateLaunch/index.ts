export{ default as CreateLaunch} from './CreateLaunch'