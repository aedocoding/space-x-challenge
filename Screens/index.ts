export {LaunchDetails} from './LaunchDetails'
export {CreateLaunch} from './CreateLaunch'
export {LaunchList} from './LaunchList'