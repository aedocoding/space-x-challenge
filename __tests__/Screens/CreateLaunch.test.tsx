import renderer from "react-test-renderer";
import { ScreenType } from "../../Types/types";
import { CreateLaunch } from "../../Screens";
jest.useFakeTimers();

describe("<CreateLaunch />", () => {
  const mockNavigatorFunction = (param: ScreenType) => {};
  it("has 1 child", () => {
    const tree = renderer
      .create(<CreateLaunch navigatorFunction={mockNavigatorFunction} />)
      .toJSON();
    expect(tree.children.length).toBe(4);
  });
  test('renders correctly', () => {
    const tree = renderer.create(<CreateLaunch navigatorFunction={mockNavigatorFunction} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
