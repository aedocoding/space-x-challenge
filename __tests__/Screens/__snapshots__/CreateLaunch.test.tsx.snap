// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateLaunch /> renders correctly 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "padding": 5,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View
        style={
          Object {
            "borderWidth": 1,
            "height": 175,
            "marginTop": 50,
            "width": 175,
          }
        }
      />
      <Text>
        Click to add
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderWidth": 1,
        "marginVertical": 10,
        "padding": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      placeholder="Name..."
      rejectResponderTermination={true}
      style={Object {}}
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "borderWidth": 1,
        "marginVertical": 10,
        "padding": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      placeholder="Description..."
      rejectResponderTermination={true}
      style={Object {}}
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <Text>
        Back
      </Text>
    </View>
    <View
      style={
        Object {
          "borderWidth": 1,
          "flex": 4,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
